main.js

```diff
@@ -1,136 +1,141 @@
+    MXN: "mx", // Mexican Peso
+    ZAR: "za", // South African Rand
+    RUB: "ru", // Russian Rubles
+    THB: "th", // Thai Baht
+    SEK: "se", // Swedish Krona
+    NOK: "no", // Norwegian Krone
+    DKK: "dk", // Danish Krone
+    MYR: "my", // Malaysian Ringgit
+    IDR: "id", // Indonesian Rupiah
+    PHP: "ph", // Philippine Peso
+    PLN: "pl", // Polish Złoty
+    BGN: "bg", // Bulgarian Lev
+    CZK: "cz", // Czech Koruna
+    HUF: "hu", // Hungarian Forint
+    ILS: "il", // Israeli New Sheqel
+    ISK: "is", // Icelandic Króna
+    RON: "ro", // Romanian Leu
+    TRY: "tr"  // Turkish Lira
 };
 
 const baseCurrencySelect = document.getElementById('baseCurrency');
 const targetCurrencySelect = document.getElementById('targetCurrency');
 const baseFlagElement = document.getElementById('baseFlag');
 const targetFlagElement = document.getElementById('targetFlag');
+const amountInput = document.getElementById('amount');
+const currentRateElement = document.getElementById('currentRateElement');
+const convertedAmount = document.getElementById('convertedAmount');
 
 async function fetchCurrencies() {
     try {
         const response = await fetch("https://api.frankfurter.dev/v1/currencies");
         const currencies = await response.json();
-        return currencies;
+        populateDropdowns(currencies);
+        return;
     } catch (error) {
         console.error("Unable to retrieve data:", error);
         return {}; 
     }
 }
 
-async function populateDropdowns() {
-    const currencies = await fetchCurrencies();
-
-    if (Object.keys(currencies).length === 0) {
-        console.error("No data");
-        return;
+let originalOptions = [];
+
+function populateDropdowns(currencies) {   
+    originalOptions = []; 
+    for (const code in currencies) {
+        const name = currencies[code];
+        const option = document.createElement('option');
+        option.value = code;
+        option.textContent = `${code} - ${name}`;
+        originalOptions.push(option);
+        baseCurrencySelect.appendChild(option.cloneNode(true));
+        targetCurrencySelect.appendChild(option.cloneNode(true));
     }
 
-    for (const [code, name] of Object.entries(currencies)) {
-        const option1 = document.createElement('option');
-        const option2 = document.createElement('option');
-
-        option1.value = code;
-        option1.textContent = `${code} - ${name}`;
-        option2.value = code;
-        option2.textContent = `${code} - ${name}`;
-
-        baseCurrencySelect.appendChild(option1);
-        targetCurrencySelect.appendChild(option2);
-    }
-
-    baseCurrencySelect.value = "EUR"; 
-    targetCurrencySelect.value = "BRL"; 
+    baseCurrencySelect.value = "EUR";
+    targetCurrencySelect.value = "BRL";
 
+    updateDropdownOptions(); 
     updateFlag(baseCurrencySelect, baseFlagElement);
     updateFlag(targetCurrencySelect, targetFlagElement);
 
     updateChart();
 }
 
+function updateDropdownOptions() {
```

Comment:
This is the function responsible for the Dropdown Update context, but it is just too big